components/org.wso2.carbon.identity.application.authenticator.hypr/src/main/java/org/wso2/carbon/identity/application/authenticator/hypr/web/HYPRAuthorizationAPIClient.java

```diff
@@ -0,0 +1,161 @@
+        this.objectMapper = new ObjectMapper();
+        this.objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    }
+
+    public String getBaseUrl() {
+        return this.hyprConfiguration.get(HyprAuthenticatorConstants.HYPR.BASE_URL);
+    }
+
+    public String getAppId() {
+        return this.hyprConfiguration.get(HyprAuthenticatorConstants.HYPR.APP_ID);
+    }
+
+    public String getApiToken() {
+        return this.hyprConfiguration.get(HyprAuthenticatorConstants.HYPR.HYPR_API_TOKEN);
+    }
+
+    public Map<String, String> getHyprConfiguration() {
+        return this.hyprConfiguration;
+    }
+
+    /**
+     * Call the HYPR server API to retrieve the registered devices.
+     *
+     * @param username          The username provided by the user.
+     * @return response         A HTTPResponse object.
+     * @throws HYPRAuthnFailedException
+     * */
+    public HttpResponse getRegisteredDevicesRequest (String username) throws HYPRAuthnFailedException {
+        //{{baseUrl}}/rp/ api/oob/client/authentication/{{appId}}/{{username}}/devices
+
+
+            String deviceInfoURL = String.format("%s%s%s/%s/devices", this.getBaseUrl(),
+                    HyprAuthenticatorConstants.HYPR.HYPR_USER_DEVICE_INFO_PATH, this.getAppId(), username);
+
+            try {
+                HttpResponse response = HYPRWebUtils.jsonGet(this.getHyprConfiguration(), deviceInfoURL);
+
+                if (response.getStatusLine().getStatusCode() > 400) {
+                        throw new HYPRAuthnFailedException(
+                                HyprAuthenticatorConstants.ErrorMessages.
+                                        AUTHENTICATION_FAILED_RETRIEVING_REG_DEVICES_FAILURE.getCode(),
+                                HyprAuthenticatorConstants.ErrorMessages.
+                                        AUTHENTICATION_FAILED_RETRIEVING_REG_DEVICES_FAILURE.getMessage());
+                }
+
+                return response;
+
+            } catch (IOException e) {
+                throw new HYPRAuthnFailedException(
+                        HyprAuthenticatorConstants.ErrorMessages
+                                .AUTHENTICATION_FAILED_RETRIEVING_REG_DEVICES_FAILURE.getCode(),
+                        HyprAuthenticatorConstants.ErrorMessages.
+                                AUTHENTICATION_FAILED_RETRIEVING_REG_DEVICES_FAILURE.getMessage(), e);
+            }
+    }
+
+    /**
+     * Call the HYPR server API to initiate the authentication request via sending the push notification to the
+     * registered device.
+     *
+     * @param namedUser          The username provided by the user.
+     * @param machineId          The machineId unique per user.
+     * @return response         A HTTPResponse object.
+     * @throws HYPRAuthnFailedException
+     * */
+    public HttpResponse initiateAuthenticationRequest(String namedUser, String machineId)
+            throws HYPRAuthnFailedException {
+        try {
+
```

Comment:
Don't we need to check null or empty for namedUser and  machineId

Reply:
Fixed with [95b498c](https://github.com/wso2-extensions/identity-outbound-auth-hypr/pull/1/commits/95b498cd32687e7b5c7d9ce0e16c99c2b9ceef02)